### PR TITLE
TEXT-202 - Extract duplicate code

### PR DESCRIPTION
**Pull request by arturobernalg**

### CODE REVIEW

File: src/main/java/org/apache/commons/text/StrBuilder.java [hunk 1]
```diff
@@ -319,15 +319,14 @@ public StrBuilder append(final boolean value) {
...
-            buffer[size++] = 'e';
         } else {
             ensureCapacity(size + 5);
             buffer[size++] = 'f';
             buffer[size++] = 'a';
             buffer[size++] = 'l';
             buffer[size++] = 's';
-            buffer[size++] = 'e';
         }
+        buffer[size++] = 'e';
```
Comment (kinow) [inline src/main/java/org/apache/commons/text/StrBuilder.java:329]: I prefer the other version, even though this one indeed reduces the duplication, the other version clearly shows that we have `true` and `false`, without the dev having to think why `tru` and `fals`, just to find the `e` was set out the `if/else` statement.

**Review by kinow — CHANGES_REQUESTED** [review]

File: src/main/java/org/apache/commons/text/StrBuilder.java (hunk 1 above)
Comment (arturobernalg) [reply src/main/java/org/apache/commons/text/StrBuilder.java]: HI @kinow 
sound fair. Changed
TY

Comment (garydgregory) [reply src/main/java/org/apache/commons/text/StrBuilder.java]: I agree with @kinow, you have to think about maintainers reading the code, not just applying changes mechanically.

Comment (arturobernalg) [reply src/main/java/org/apache/commons/text/StrBuilder.java]: Yep. Agree.

File: src/main/java/org/apache/commons/text/StrTokenizer.java [hunk 2]
```diff
@@ -842,10 +838,9 @@ private int readWithQuotes(final char[] srcChars, final int start, final int len
                     continue;
                 }
 
-                // copy regular character from outside quotes
-                workArea.append(srcChars[pos++]);
-                trimStart = workArea.size();
             }
+            workArea.append(srcChars[pos++]);
```
Comment (kinow) [inline src/main/java/org/apache/commons/text/StrTokenizer.java:842]: @arturobernalg can you add back the comment above these lines, please?

File: src/main/java/org/apache/commons/text/StringTokenizer.java [hunk 3]
```diff
@@ -848,11 +844,9 @@ private int readWithQuotes(final char[] srcChars, final int start, final int len
                     pos += trimmedLen;
                     continue;
                 }
-
-                // copy regular character from outside quotes
-                workArea.append(srcChars[pos++]);
-                trimStart = workArea.size();
             }
+            workArea.append(srcChars[pos++]);
```
Comment (kinow) [inline src/main/java/org/apache/commons/text/StringTokenizer.java:848]: Ditto here about the comment that was removed :+1:

File: src/main/java/org/apache/commons/text/StrTokenizer.java (hunk 2 above)
Comment (arturobernalg) [reply src/main/java/org/apache/commons/text/StrTokenizer.java:843]: changed

File: src/main/java/org/apache/commons/text/StringTokenizer.java (hunk 3 above)
Comment (arturobernalg) [reply src/main/java/org/apache/commons/text/StringTokenizer.java:849]: changed. TY

**Review by kinow — APPROVED** [review]

**Comment by kinow** [comment]
Merged, thanks!

**Pull request closed without merging**